lib/internal/modules/esm/loader.js

```diff
@@ -20,9 +20,14 @@ const {
 } = require('internal/modules/esm/utils');
 let defaultResolve, defaultLoad, importMetaInitializer;
 
-function newModuleMap() {
-  const ModuleMap = require('internal/modules/esm/module_map');
-  return new ModuleMap();
+function newLoadCache() {
+  const { LoadCache } = require('internal/modules/esm/module_map');
+  return new LoadCache();
+}
+
+function newResolveCache() {
+  const { ResolveCache } = require('internal/modules/esm/module_map');
+  return new ResolveCache();
 }
```

Comment:
Since `resolve` happens first, I feel like we should define its cache first.

```suggestion
function newResolveCache() {
  const { ResolveCache } = require(‘internal/modules/esm/module_map’);
  return new ResolveCache();
}

function newLoadCache() {
  const { LoadCache } = require(‘internal/modules/esm/module_map’);
  return new LoadCache();
}
```